### PR TITLE
Deprecate abort_on_alloc_failure

### PR DESCRIPTION
**Pull request by kaidokert**
This config is unused and always true.

b/150410605

### CODE REVIEW

File: cobalt/site/docs/reference/starboard/gn-configuration.md [hunk 1]
```diff
@@ -5,7 +5,6 @@ Book: /youtube/cobalt/_book.yaml
 
 | Variables |
 | :--- |
-| **`abort_on_allocation_failure`**<br><br> Halt execution on failure to allocate memory.<br><br>The default value is `true`. |
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply cobalt/site/docs/reference/starboard/gn-configuration.md:8]: The original code was added in go/cobalt-cl/52783 - the use case wasn't specified as far as i can tell. The config flag seems to be added to provide for gradual platform-by-platform migration to checked allocators, but that migration never fully concluded. As far as i can trace, we have simply copied it through the gyp migrations without ever making recommendations around its usage.

All our own platforms leave it on in every build and every config. It is possible that some partners have flipped it. That would make their builds and platforms behave quite a bit differently than our references, and memory allocation related crashes would get more obscured ( e.g. stack trace wouldn't necessarily point at alloc failure, but some next pointer usage down the road ).

Differing behaviors at that low level are generally against our goals of more uniform and testable stack. Also, if a partner really needs something similar - this is in Starboard implementation, it does not affect the interface, they can always remove the checks on their own.

I think i did forget to add starboard/CHANGELOG.md here, though

[the remark this answers is not in the document]
Comment (kaidokert) [reply cobalt/site/docs/reference/starboard/gn-configuration.md:8]: Changelog entry added as well.

File: starboard/shared/starboard/memory.cc [hunk 2]
```diff
@@ -102,27 +102,15 @@ void* SbMemoryAllocateChecked(size_t size) {
 namespace {
```
Comment (y4vor) [inline starboard/shared/starboard/memory.cc:103]: SbMemoryAllocated is deprecated in SB16.

Comment (kaidokert) [reply starboard/shared/starboard/memory.cc:103]: Right - so this change is a no-op. Are you suggesting i leave the #ifdefs in here and just remove the config from gn/md files ?

**Review by hlwarriner — APPROVED** [review]

File: starboard/shared/starboard/memory.cc (hunk 2 above)
Comment (y4vor) [reply starboard/shared/starboard/memory.cc:103]: I guess the question is do you want to potentially affect SB15/SB14 implementations if partners rebase and rebuild, but don't change the SB version?

**Review by y4vor — APPROVED** [review]

File: starboard/shared/starboard/memory.cc (hunk 2 above)
Comment (kaidokert) [reply starboard/shared/starboard/memory.cc]: I backed out the .cc changes on the off chance that someone somewhere used it. It had no effect in our build configs.

**Pull request merged**